src/agentlab/agents/generic_agent/generic_agent.py

```diff
@@ -1,27 +1,33 @@
-import traceback
 from dataclasses import asdict, dataclass
 from functools import partial
 from warnings import warn
 
 from browsergym.experiments.agent import Agent
-from browsergym.experiments.loop import AbstractAgentArgs
 from langchain.schema import HumanMessage, SystemMessage
 
 from agentlab.agents import dynamic_prompting as dp
 from agentlab.agents.utils import openai_monitored_agent
 from agentlab.llm.chat_api import BaseModelArgs
-from agentlab.llm.llm_utils import ParseError, RetryError, retry_raise
+from agentlab.llm.llm_utils import RetryError, retry_raise
+from agentlab.agents.agent_args import AgentArgs
 
 from .generic_agent_prompt import GenericPromptFlags, MainPrompt
 
 
 @dataclass
-class GenericAgentArgs(AbstractAgentArgs):
-    agent_name: str = "GenericAgent"
+class GenericAgentArgs(AgentArgs):
     chat_model_args: BaseModelArgs = None
     flags: GenericPromptFlags = None
     max_retry: int = 4
 
+    @property
+    def agent_name(self):
+        return f"GenericAgent-{self.chat_model_args.model_name}"
```

Comment:
this is going to cause some issues, bc some model_names have '/' in their name (eg huggingface models).
we should either add .replace('/', '_'), or make some weird updates to the naming convention in llm_configs.py